app/hooks/use-fetch-fee-rate.ts

```diff
@@ -0,0 +1,24 @@
+import { useCallback, useState, useEffect } from 'react';
+import BigNumber from 'bignumber.js';
+
+import { useApi } from './use-api';
+
+export function useFetchFeeRate() {
+  const [feeRate, setFeeRate] = useState(1);
+
+  const api = useApi();
+
+  useEffect(() => {
+    async function run() {
+      const resp = await api.getFeeRate();
+      setFeeRate(new BigNumber(resp.data).toNumber());
+    }
+    void run();
+  }, [api]);
+
+  // use react-query
+  const calculateFee = useCallback((bytes: number) => new BigNumber(feeRate).multipliedBy(bytes), [
```

Comment:
small nit, i would not expect this kind of side effect/method to be exposed from a `useFetchX` hook.

Reply:
Thought you'd pick up on this lol. Totally agree, only reason I've done this is 'cause the react-query stuff is in another PR I didn't bother to rebase onto.

Reply:
@aulneau refactored to separate hook

```ts
const defaultFeeRate = 1;

export function useCalculateFee() {
  const { feeRate } = useFetchFeeRate();
  return useCallback(
    (bytes: number) => new BigNumber(feeRate ?? defaultFeeRate).multipliedBy(bytes),
    [feeRate]
  );
}
```